package/MDAnalysis/analysis/parallel.py

```diff
@@ -0,0 +1,375 @@
+        from multiprocessing import Pool
+
+        with Pool(processes=self.n_workers) as pool:
+            results = pool.map(func, computations)
+        return results
+
+
+class BackendDask(BackendBase):
+    """A built-in backend that executes a given function using dask.delayed.compute method with `scheduler='processes'`
+    and `chunksize=1`. Requires `dask` module to be installed.
+
+    .. versionadded: 2.7.0
+    """
+
+    def apply(self, func: Callable, computations: list) -> list:
+        from dask.delayed import delayed
+        import dask
+
+        computations = [delayed(func)(task) for task in computations]
+        results = dask.compute(computations, scheduler="processes", chunksize=1, n_workers=self.n_workers)[0]
+        return results
+
+    def _get_checks(self):
+        base_checks = super()._get_checks()
+        checks = {is_installed("dask"): "module 'dask' should be installed: run 'python3 -m pip install dask'"}
+        return base_checks | checks
+
+
+class ResultsGroup:
+    """
+    Management and aggregation of results stored in :class:`Result` instances.
+
+    A :class:`ResultsGroup` is an optional description for :class:`Result` "dictionaries"
+    that are used in analysis classes based on :class:`AnalysisBase`. For each *key* in a
+    :class:`Result` it describes how multiple pieces of the data held under the key are
+    to be aggregated. This approach is necessary when parts of a trajectory are analyzed
+    independently (e.g., in parallel) and then need to me merged (with :meth:`merge`) to
+    obtain a complete data set.
+
+    Parameters
+    ----------
+    lookup : dict[str, Callable], optional
+        aggregation functions lookup dict, by default None
+
+    Examples
+    --------
+    >>> from MDAnalysis.analysis.parallel import ResultsGroup, Results
+    >>> group = ResultsGroup(lookup={'mass': ResultsGroup.float_mean})
+    >>> obj1 = Results(mass=1)
+    >>> obj2 = Results(mass=3)
+    >>> group.merge([obj1, obj2])
+    {'mass': 2.0}
+
+    >>> # you can also set `lookup[attribute]=None` to those attributes that you want to skip
+    >>> lookup = {'mass': ResultsGroup.float_mean, 'trajectory': None}
+    >>> group = ResultsGroup(lookup)
+    >>> objects = [Results(mass=1, skip=None), Results(mass=3, skip=object)]
+    >>> group.merge(objects, require_all_aggregators=False)
+    {'mass': 2.0}
+
+    .. versionadded: 2.7.0
+    """
+
+    def __init__(self, lookup: dict[str, Callable] = None):
+        self._lookup = lookup
+
+    def merge(self, objects: Sequence[Results], require_all_aggregators: bool = True) -> Results:
+        """Merge results into a single object. If objects contain single element, returns it while ignoring _lookup attribute.
+
+        Parameters
+        ----------
+        require_all_aggregators : bool, optional
+            if you want to raise an exception when no aggregation function for a particular argument is found, by default True.
```

Comment:
```suggestion
            if you want to raise an exception when no aggregation function for a particular argument is found
```
Not sure if we need to state the default here. It's already in the function signature, isn't it? If changed, this piece of documentation might easily diverge.